src/directory_listing.rs

```diff
@@ -303,14 +309,15 @@ async fn read_dir_entries(
             FileType::File
         };
 
-        file_entries.push(FileEntry {
+        let uri = utf8_percent_encode(&uri, PERCENT_ENCODE_SET).to_string();
```

Comment:
This is a bug, you are encoding `base_dir` twice. Please try listing a directory with non-alphanumeric characters in the name.

Reply:
Yeah, I was too lazy looks like. So reverted that for now. This can be better simplified later.